### PR TITLE
Change condition checking in test_line

### PR DESCRIPTION
**Pull request by szymag**
Before that, these tests were checked if objects are the same with `==`
operator, not with geometrical entity's `equals()` method.

For example:
Objects `Line((0,0,0), (1,1,1))` and `Line((0,0,0), (2,2,2))` represent the
same line, but they are not the same object, so `==` comparison returned
`False`, but when we use `equals()` we get `True`.
Previously, all tests passed, because only trivial example were considered.

### CODE REVIEW

File: sympy/geometry/tests/test_line.py [hunk 1]
```diff
@@ -54,13 +54,13 @@ def test_line_geom():
     raises(ValueError, lambda: Line(Point(0, 0), Point(0, 0)))
 
     # Basic stuff
-    assert Line((1, 1), slope=1) == Line((1, 1), (2, 2))
```
Comment (smichr) [inline sympy/geometry/tests/test_line.py]: My concern here is that the use of `==` affirms the form/arguments the result for a Line instantiated with a keyword should have. The use of `equals` would permit a divergence from that. Perhaps (if `equals` is not tested well) a few tests to improve coverage alone would be a better improvement to the testing file.

Comment (szymag) [inline sympy/geometry/tests/test_line.py]: Thank you for your comment. You are probably right, that adding `equals()` calls separately would be better. 
I will revise my changes and I will leave these `equals()` calls, which really improve coverage. For example this usage of `equals` seems quite reasonable for me (of course two asserts are valuable), because it checks mathematical identity, which in this case is not implied by `==` equality. The objects are different but they can differ because of different reasons:
```python 
-    assert l1 != l2
+    assert l1.equals(l2)
```

BTW: I'm going to take a look on non-test code now, but I'm thinking also about granulating this file - I mean many different test cases are hidden under `test_line_geom` name and they could be split into many smaller functions, which would improve a readability (e.g. some variables are assigned many times and it's hard to follow the code). What do you think about it?

Comment (smichr) [inline sympy/geometry/tests/test_line.py]: I would spend time on it if its something that I had to look at a lot. But I don't so I wouldn't. The most important thing is that the code if covered in a meaningful way. Beautifying the test suite is low on my list of things to do. But it may be meaningful to you if it is an entry way to better understanding the geometry module and I would be willing to review whatever you do.
